From c8ae68c59990fd132f673ede1dfda9c3188af488 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Thu, 6 Apr 2017 15:20:18 +0300
Subject: [PATCH 0951/1345] fix: dts: a70x0: relocate A7040-DB specific IO
 expander interrupt to configuration D

commit  5386567517a3516203280f9ec120ae70890edf91 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

IO expander interrupt is connected to MPP7 on
armada-7040-DB for setup D only,
while on setup A,B,C,E its connected to MPP62.
MPP62 is used for SD/MMC for setups A,B,C,E,
hence can't be used for IO expander interrupt.
Therefore these changes should be in D configuration
device tree and not in the 7040-db.dtsi.

Change-Id: Iac189c242c9ba2e5defc690fe52afec92d5283f1
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38557
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts |   22 ++++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi  |   20 --------------------
 2 files changed, 22 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
index 7a58664..7c8bcd2 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
@@ -73,6 +73,13 @@
 
 	cpn-110-master {
 		config-space {
+			pinctrl@440000 {
+			compatible = "marvell,a70x0-pinctrl";
+				pca0_pins: pca0_pins {
+					marvell,pins = "mpp7";
+					marvell,function = "gpio";
+				};
+			};
 			serial@702000 {
 				status = "disabled";
 			};
@@ -104,6 +111,7 @@
 				status = "okay";
 			};
 			u3d@520000 {
+				vbus-gpio = <&expander0 3 GPIO_ACTIVE_HIGH>;
 				status = "okay";
 			};
 			udc@524100 {
@@ -113,6 +121,11 @@
 				status = "okay";
 			};
 			sdhci@780000 {
+				/* Support SD detection in Linux, disabled for
+				 * now
+				 * cd-gpios = <&expander0 12 GPIO_ACTIVE_HIGH>;
+				 * pinctrl-0 = <&cpm_sdhci_pins>;
+				 */
 				status = "okay";
 			};
 			spi@700680 {
@@ -121,6 +134,15 @@
 			i2c1: i2c@701000 {
 				status = "okay";
 				clock-frequency = <100000>;
+
+				expander0: pca9555@21 {
+					pinctrl-0 = <&pca0_pins>;
+					/* GPIO 62 is defined as GPIO 30 of cpm_gpio1 */
+					interrupt-parent = <&cpm_gpio0>;
+					interrupts = <7 IRQ_TYPE_EDGE_FALLING>;
+					interrupt-controller;
+					#interrupt-cells = <2>;
+				};
 			};
 			i2c2: i2c@701100 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 3366856..093778d 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -76,13 +76,6 @@
 
 	cp110-master {
 		config-space@f2000000 {
-			pinctrl@440000 {
-			compatible = "marvell,a70x0-pinctrl";
-				pca0_pins: pca0_pins {
-					marvell,pins = "mpp62";
-					marvell,function = "gpio";
-				};
-			};
 			i2c1: i2c@701000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&i2c0_pins>;
@@ -99,22 +92,11 @@
 				expander0: pca9555@21 {
 					compatible = "nxp,pca9555";
 					pinctrl-names = "default";
-					pinctrl-0 = <&pca0_pins>;
-					interrupt-parent = <&cpm_gpio1>;
-					/* GPIO 62 is defined as GPIO 30 of cpm_gpio1 */
-					interrupts = <30 IRQ_TYPE_EDGE_FALLING>;
 					gpio-controller;
 					#gpio-cells = <2>;
-					interrupt-controller;
-					#interrupt-cells = <2>;
 					reg = <0x21>;
 				};
 			};
-			u3d@520000 {
-				pinctrl-0 = <&u3d_vbus_pins>;
-				pinctrl-names = "default";
-				vbus-gpio = <&cpm_gpio1 30 GPIO_ACTIVE_HIGH>;
-			};
 			usb3h0: usb3@500000 {
 				usb-phy = <&usb3h0_phy>;
 			};
@@ -123,9 +105,7 @@
 			};
 			sdhci@780000 {
 				pinctrl-names = "default";
-				pinctrl-0 = <&cpm_sdhci_pins>;
 				bus-width = <4>;
-				non-removable;
 			};
 			spi@700680 {
 				spi-flash@0 {
-- 
1.7.9.5

